appnav/src/main/kotlin/io/element/android/appnav/di/MatrixClientsHolder.kt

```diff
@@ -29,13 +30,14 @@ private const val SAVE_INSTANCE_KEY = "io.element.android.x.di.MatrixClientsHold
 @ContributesBinding(AppScope::class)
 class MatrixClientsHolder @Inject constructor(
```

Comment:
Maybe we should rename this class so it's a bit clearer it's not just about `MatrixClients`?
Also we might want to create a class instead of using Pair? 

I don't think we need the `DefaultSyncOrchestratorProvider`, we should let `MatrixClientsHolder` (with a new name :P)  implements the `SyncOrchestratorProvider`